pkg/graphql/request.go

```diff
@@ -2,17 +2,40 @@ package graphql
 
 import (
 	"encoding/json"
+	"errors"
 	"io"
+	"io/ioutil"
 )
 
+var ErrEmptyRequest = errors.New("the provided request is empty")
+
 type Request struct {
 	OperationName string          `json:"operation_name"`
 	Variables     json.RawMessage `json:"variables"`
 	Query         string          `json:"query"`
 }
 
 func UnmarshalRequest(reader io.Reader) (*Request, error) {
-	return &Request{}, nil
+	requestBytes, err := ioutil.ReadAll(reader)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(requestBytes) == 0 {
+		return nil, ErrEmptyRequest
+	}
+
+	var request Request
+	err = json.Unmarshal(requestBytes, &request)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(request.Query) == 0 {
```

Comment:
Should Unmarshal also validate the request content?

Reply:
valid point - this should happen in the validation